### PR TITLE
Add documentation menu item referencing internal `Find Rules` page

### PR DESCRIPTION
**Pull request by tigitz**
Related to https://github.com/rectorphp/getrector-com/issues/2512

I've introduced a generic view object for the documentation menu item to seamlessly handle both internal link types and documentation section types.

Could handle external link too if needed at some point.

### CODE REVIEW

File: resources/views/docs/section.blade.php [hunk 1]
```diff
@@ -12,32 +12,35 @@
...
                                 &nbsp;
                             @endif
 
-                            <a href="{{ action(\App\Controller\DocumentationController::class, ['section' => $documentation_section->getSlug()]) }}"
-                               class="{{ $documentation_section->isNew() ? 'text-bold' : '' }}"
+                            <a href="{{ $documentation_menu_item->getHref() }}"
+                               class="{{ $documentation_menu_item->isNew() ? 'text-bold' : '' }}"
+                               @if($documentation_menu_item->mustOpenInNewWindow())
+                                   target="_blank"
+                               @endif
```
Comment (TomasVotruba) [inline resources/views/docs/section.blade.php:41]: This is not needed, as in control of user.

File: src/Documentation/DocumentationMenuFactory.php [hunk 2]
```diff
@@ -4,47 +4,72 @@
...
-                new DocumentationSection('define-paths', 'Define Paths'),
-                new DocumentationSection('set-lists', 'Set Lists'),
-                new DocumentationSection('ignoring-rules-or-paths', 'Ignoring Rules or Paths'),
-                new DocumentationSection('import-names', 'Import Names'),
-                new DocumentationSection('configured-rules', 'Configured Rules'),
+                $this->documentationMenuItemFactory->createSection('integration-to-new-project', 'New Project', true),
+                $this->documentationMenuItemFactory->createSection('define-paths', 'Define Paths'),
+                $this->documentationMenuItemFactory->createSection('set-lists', 'Set Lists'),
+                $this->documentationMenuItemFactory->createInternalLink(
+                    FilterRectorController::class,
```
Comment (TomasVotruba) [inline src/Documentation/DocumentationMenuFactory.php:30]: :+1:

**Comment by tigitz** [comment]
@TomasVotruba ready to be reviewed :slightly_smiling_face:

**Comment by TomasVotruba** [comment]
Awesome, thank you :+1:

**Pull request merged**